src/Models/Test/ModelInstantiatorTest.php

```diff
@@ -434,4 +435,57 @@ public function testDontConvertNullPropertyOnRetrieve()
             )
         );
     }
+
+    public function testDatabaseModelToDataObjectNullableNestedObjectNotNullWithRelationsGetter(): void
+    {
+        // Verify a nullable one to many relationship works when the data is defined.
+        $db_model = new ModelInstantiatorTestDBModel(123);
+        $relation = new ModelInstantiatorTestDBModel(456);
+
+        $db_model->addNullableOneRelation($relation);
+
+        $data_object = $this->model_instantiator->databaseModelToDataObject(
+            $db_model,
+            ModelInstantiatorTestObject::class,
+        );
+
+        $this->assertEquals($data_object->nullableFakeHasOneRelation->pdgaNumber, 456);
+    }
+
+    public function testDatabaseModelToDataObjectNullableNestedObjectNullWithRelationsGetter(): void
+    {
+        // Verify a nullable one to many relationship works when the data is null.
+        $db_model = new ModelInstantiatorTestDBModel(123);
+        $relation = null;
+
+        $db_model->addNullableOneRelation($relation);
+
+        $data_object = $this->model_instantiator->databaseModelToDataObject(
+            $db_model,
+            ModelInstantiatorTestObject::class,
+        );
+
+        $this->assertEquals($data_object->nullableFakeHasOneRelation, null);
+    }
+
+    public function testDatabaseModelToDataObjectNestedObjectWithRelationsGetterExcepts(): void
+    {
+        // Verify a one to many relationship errors out with the correct exception type when the data is null.
+        $db_model = new ModelInstantiatorTestDBModel(123);
+        $relation = null;
+
+        $db_model->addOneRelation($relation);
+
+        try
+        {
+            $data_object = $this->model_instantiator->databaseModelToDataObject(
+                $db_model,
+                ModelInstantiatorTestObject::class,
+            );
```

Comment:
This _pattern_ of testing for an exception may appear to pass if the expected exception isn't actually raised, because, in that case, the `try` block will complete successfully.  To show that the `catch` block executes, add `$this->assertTrue(false)` here.